src/main/java/seedu/address/logic/parser/AddServiceCommandParser.java

```diff
@@ -0,0 +1,49 @@
+package seedu.address.logic.parser;
+
+import static seedu.address.commons.core.Messages.MESSAGE_INVALID_COMMAND_FORMAT;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_COST;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_DESCRIPTION;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_ID;
+
+import java.util.stream.Stream;
+
+import seedu.address.logic.commands.AddServiceCommand;
+import seedu.address.logic.parser.exceptions.ParseException;
+import seedu.address.model.hotel.bill.Cost;
+import seedu.address.model.ids.AvailableServiceId;
+
+/**
+ *  Parses input arguments ad creates new AddServiceCommand object.
+ */
+public class AddServiceCommandParser {
+    /**
+     * Parses the given {@code String} of arguments in the context of the AddServiceCommand
+     * and returns an AddServiceCommand object for execution.
+     * @throws ParseException if the user input does not conform the expected format
+     */
+    public AddServiceCommand parse(String args) throws ParseException {
+        ArgumentMultimap argMultimap =
+                ArgumentTokenizer.tokenize(args, PREFIX_DESCRIPTION, PREFIX_COST, PREFIX_ID);
+
+        if (!arePrefixesPresent(argMultimap, PREFIX_DESCRIPTION, PREFIX_COST, PREFIX_ID)
```

Comment:
Code seems to require user to input `i/ID` but AddServiceCommand states that id is automatically generated?

Reply:
Ah, I rethink about it and see it's almost impossible to generate these Ids properly.

Reply:
also, it's just syntax and pattern recognizer. I think it's okay, just for parsing commands.

Reply:
Hmm then do we still charge the services by service id? (If not i think this command might be of not much use)

If we are I suggest asking for user input -> new prefix + update command usage message!

Reply:
yes, but I mean /i is just used for parsing, not really affect the functioning. And the hotel will show a list of services with ID for the receptionist to refer to. If not, he/she will need to remember all the costs of all services.